domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading.Tasks;
+using API.Authorization._Features_;
+using Asp.Versioning;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Authorization.Controllers;
+
+[ApiController]
+[ApiVersion(ApiVersions.Version20230101)]
+[Route("api/authorization/terms")]
+public class TermsController(IMediator mediator) : ControllerBase
+{
+    [ProducesResponseType(typeof(TermsResponseDto), StatusCodes.Status200OK)]
+    [HttpPost("accept")]
+    public async Task<IActionResult> AcceptTerms([FromBody] AcceptTermsDto acceptTermsDto)
+    {
+        var isValid = await mediator.Send(new OrganizationStateQuery(acceptTermsDto.Tin));
+
+        if (isValid)
+            return Ok(new TermsResponseDto(true));
+
+        var latestTerms = await mediator.Send(new GetLatestTermsQuery());
+
+        if (acceptTermsDto.TermsVersion != latestTerms.Version)
+            return Ok(new TermsResponseDto(false, latestTerms.Text, latestTerms.Version));
+
+        var createResult = await mediator.Send(new CreateOrganizationAndUserCommand(
+            acceptTermsDto.Tin,
+            acceptTermsDto.OrganizationName,
+            acceptTermsDto.UserIdpId,
+            acceptTermsDto.UserName,
+            latestTerms.Version
+        ));
+
+        return Ok(new TermsResponseDto(true, CreateResult: createResult));
+    }
+}
+
+public record AcceptTermsDto(
+    string Tin,
```

Comment:
All except accept terms can be taken from Claims. You shouldn't be able to accept on other organizations that the one your logged in as.